ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerTask.java

```diff
@@ -122,6 +122,22 @@ public void run() {
                     // execution until it is complete. See JIRA AS7-3119
                     if (timer.getState() == TimerState.IN_TIMEOUT || timer.getState() == TimerState.RETRY_TIMEOUT) {
                         EJB3_TIMER_LOGGER.skipOverlappingInvokeTimeout(timer, now);
+
+                        if (EJB3_TIMER_LOGGER.isDebugEnabled()) {
+                            // WFLY-10542 log thread stack trace which is processing timer task in debug level to diagnose timer overlap
+                            Thread otherThread = timer.getExecutingThread();
+                            // can be null for clustered timers if the timer is executing on another node.
+                            if (otherThread != null) {
+                                String threadBasic = "Thread: " + otherThread.getName() + " Id: " + otherThread.getId() + " of group " + otherThread.getThreadGroup() + " is in state: " + otherThread.getState();
+                                StringBuffer debugMsg = new StringBuffer();
+                                debugMsg.append(threadBasic + System.lineSeparator());
+                                for (StackTraceElement ste : otherThread.getStackTrace()) {
+                                    debugMsg.append(ste.toString() + System.lineSeparator());
+                                }
+                                EJB3_TIMER_LOGGER.debugf(debugMsg.toString());
```

Comment:
This is a bit nitpicky especially since this is a debug log message, but something like this would be more efficient.
```
final StringBuilder debugMsg = new StringBuilder()
        .append("Thread: ")
        .append(otherThread.getName())
        .append(" Id: ")
        .append(otherThread.getId())
        .append(" of group ")
        .append(otherThread.getThreadGroup())
        .append(" is in state: ")
        .append(otherThread.getState());
for (StackTraceElement ste : otherThread.getStackTrace()) {
    debugMsg.append(System.lineSeparator())
            .append(ste.toString());
}
EJB3_TIMER_LOGGER.debug(debugMsg.toString());
```

Reply:
Current PR state seems to be done this way.